tests/connection/AConnectionTest.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Bolt\tests\connection;
+
+use Bolt\Bolt;
+use Bolt\connection\IConnection;
+use Bolt\connection\Socket;
+use Bolt\connection\StreamSocket;
+use Bolt\error\ConnectionTimeoutException;
+use Bolt\error\MessageException;
+use Bolt\helpers\Auth;
+use Bolt\protocol\V4;
+use PHPUnit\Framework\TestCase;
+
+final class AConnectionTest extends TestCase
+{
+    public function provideConnections(): array
+    {
+        return [
+            [StreamSocket::class],
+            [Socket::class],
+        ];
+    }
+
+    /**
+     * @dataProvider provideConnections
+     */
+    public function testMillisecondTimeout(string $alias): void
+    {
+        $socket = $this->getConnection($alias);
+        $protocol = (new Bolt($socket))->build();
+        $socket->setTimeout(1.5);
+        $protocol->init(Auth::basic($GLOBALS['NEO_USER'], $GLOBALS['NEO_PASS']));
+
+        $time = microtime(true);
+        try {
+            $protocol->run('FOREACH ( i IN range(1,10000) | 
+  MERGE (d:Day {day: i})
+)');
+        } catch (ConnectionTimeoutException $e) {
+            $newTime = microtime(true);
+
+            $this->assertEqualsWithDelta(1.5, $newTime - $time, 0.2);
+        }
+
+        self::assertTrue(true);
```

Comment:
it's neccessary if other methods doesn't have it?

Reply:
No, this was a relic from when I did not do another reset afterwards, that was my way to make sure it ran correctly. I will remove it now